app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerStateObservable.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.audio;
+
+import com.google.android.exoplayer2.ExoPlayer;
+
+import android.support.annotation.NonNull;
+
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+
+import static com.futurice.freesound.utils.Preconditions.get;
+
+/**
+ * Make an Observable from the ExoPlayer player state.
+ *
+ * Based upon techniques used in RxBinding library.
+ */
+final class ExoPlayerStateObservable extends Observable<ExoPlayerState> {
```

Comment:
My previous comment must have been lost. How do I get the dispose hook to remove the listener when using `create`?